### DISCUSSION TITLE
Duda Pokedex movimientos

### DISCUSSION THREAD

**Discussion by AndresAVargas** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#527] [Q&A]
Hola compañeros, no se si mue puedan orientar, estoy haciendo la pokedex y note que para los movimientos hay pokemon que tiene 124 otros 82 y va variando, lo que estaba haciendo era hacer una función para cada movimiento pero hay alguna manera de hacerlo para sacar todos los movimientos en una sola función?

**Comment by AngelCruzO** [comment]
Con un for y una variable dónde concatenes todos. 

for(var i=0;i<data.moves.length;i++){

movimientos += data.moves[i].name

}

Más o menos así puedes hacer, ten en cuenta que puedes imprimir en el HTML directo con el for. Ojo, el ejemplo lo hice de memoria, checa bien la poke API para acceder a la información correcta

**Reply by AndresAVargas** [reply]
la variable movimientos estaría definida? o seria una funcion como:

const Movimientos = (url) => {
    const Move = document.getElementById("Move")
    Move.innerHTML = (url)
}

**Reply by AngelCruzO** [reply]
La variable movimientos debe estar definida para concatenar todos los movimientos. En tu caso sería Move, la función que hiciste está bien, solo adecuala con el ejemplo que te puse

**Comment by RodolfoMorquecho** [comment]
Investiga el uso de ".map" con ese metodos puedes acceder a todos los movimientos (arreglos), tampoco sabia como hacerlo pero así me sugirieron y funciona bien

**Comment by PlayerOneV** [comment]
Hola, en mi caso para mostrar todos los movimientos primero los guarde y los pase a un metodo.
![image](https://user-images.githubusercontent.com/75816087/160218874-321d6ecd-fab8-4b54-a31e-24a8a854ce5a.png)

Lo que hace mi función es crear una nueva etiqueta div por cada movimiento que tenga el Pokémon, para evitar que se desborde al div padre le puse la propiedad overflow:auto; que le pone un scroll.

Es importante utilizar el remove de la segunda linea de la función para que se borren los div creados anteriormente y de esa forma no se acumulen los movimientos de los Pokémon al cambiar a otro.
